v5/v5_with_library/v5_stable.ino

```diff
@@ -2271,4 +2361,104 @@ void setBypassKey(String tempStr) {
+  println("Final URL: [" + newRequest + "]");
+  HTTPClient localRequest;
+  localRequest.begin(newRequest);
+
+  int httpResponseCode = localRequest.GET();
+  if (httpResponseCode > 0) {
+    Serial.print("HTTP Response code: ");
+    Serial.println(httpResponseCode);
+    String payload = localRequest.getString();
+    Serial.println("Response payload: " + payload);
+    sendSMS("Request sent for switch: " + switchNumber + " time: " + time +
+            " minutes");
+  } else {
+    Serial.print("Error in HTTP GET request. HTTP Response code: ");
+    Serial.println(httpResponseCode);
+    sendSMS("Request failed for switch: " + switchNumber + " time: " + time +
+            " minutes");
+  }
+
+  localRequest.end();
+}
+
+void relayControllerManager(String input) {
+  // input should be like this:
+  // on switch 1 for 10 (10 is time in minutes)
+  // or
+  // on/off switch 1
+  String switchNumber =
+      String(fetchNumber(input.substring(input.indexOf("switch") + 7, -1)));
+  String time = "";
+  if (input.indexOf("for") != -1) {
+    time = String(fetchNumber(input.substring(input.indexOf("for") + 4, -1)));
+  }
+  if (input.indexOf("on") != -1) {
+    if (time.length() > 0) {
+      relayControllerManager(switchNumber, time);
+    } else { // common case
+      relayControllerManager(switchNumber, "998");
+    }
+  } else if (input.indexOf("off") != -1) {
+    relayControllerManager(switchNumber, "-1");
+  }
+}
+
+void addError(String errorCode) {
+  if (!isIn(error_codes, errorCode)) {
+    error_codes += " " + errorCode;
+    saveItOrangePi("New error added error code {" + errorCode + "}");
+    println("new error code added: " + errorCode);
+  } else {
+    saveItOrangePi("Error occurred again error code {" + errorCode + "}");
+    println("Error code already added: " + errorCode);
+  }
+}
+
+bool isIn(String mainString, String toFind) {
+  return mainString.indexOf(toFind) != -1;
+}
+
+bool isIn(String mainString, String toFind_1, String toFind_2) {
+  return (isIn(mainString, toFind_1) || isIn(mainString, toFind_2));
+}
+
+bool isIn(String mainString, String toFind_1, String toFind_2,
+          String toFind_3) {
+  return (isIn(mainString, toFind_1) || isIn(mainString, toFind_2) ||
+          isIn(mainString, toFind_3));
+}
+
+void log(String tempMessage) {
+  alert(tempMessage);
+  saveItOrangePi(tempMessage);
+}
+
+void rise(String tempMessage, String errorCode) {
+  log(tempMessage);
+  addError(errorCode);
+}
+
+String getRTC_Time() {
+  return (String(myRTC.hour) + ":" + String(myRTC.minutes) + ":" +
+          String(myRTC.seconds) + " ___ " + String(myRTC.month) + "/" +
+          String(myRTC.date) + "/" + String(myRTC.year));
+}
+
+void updateOnlineTime(String formattedDate) {
+  // sscanf(formattedDate.c_str(), "%d-%d-%dT%d:%d:%dZ", myRTC.year,
+  // myRTC.month,
+  //        myRTC.date, myRTC.hour, myRTC.minutes, myRTC.seconds);
+  // myRTC.year = myRTC.year % 100;
```

Comment:
Please remove this unused code too, @mhamidjamil

Reply:
I have removed it in CR3.